service/src/integrationTest/kotlin/fi/espoo/evaka/invoicing/FeeDecisionIntegrationTest.kt

```diff
@@ -942,9 +946,45 @@ class FeeDecisionIntegrationTest : FullApplicationTest() {
 
         val (_, response, _) = http.post("/decisions/confirm")
             .asUser(user)
-            .jsonBody(objectMapper.writeValueAsString(draftWithFutureDates.id))
+            .jsonBody(objectMapper.writeValueAsString(listOf(draftWithFutureDates.id)))
             .response()
         assertEquals(400, response.statusCode)
+        assertEquals(
+            "feeDecisions.confirmation.tooFarInFuture",
+            JSONObject(response.body().asString("text/json")).getString("errorCode")
+        )
+    }
+
+    @Test
+    fun `confirmDrafts when decision at last possible confirmation date exists`() {
+        doReturn(1).whenever(evakaEnv).nrOfDaysFeeDecisionCanBeSentInAdvance
+        val nowDate = LocalDate.now()
+        val draftWithFutureDates = testDecisions.find { it.status == FeeDecisionStatus.DRAFT }!!
+            .copy(validDuring = DateRange(nowDate.plusDays(1), nowDate.plusYears(1)))
+        db.transaction { tx -> tx.upsertFeeDecisions(listOf(draftWithFutureDates)) }
+
+        val (_, response, _) = http.post("/decisions/confirm")
+            .asUser(user)
+            .jsonBody(objectMapper.writeValueAsString(listOf(draftWithFutureDates.id)))
+            .response()
+        assertEquals(204, response.statusCode)
+
+        asyncJobRunner.runPendingJobsSync(2 )
+
+        val (_, _, result) = http.get("/decisions/${draftWithFutureDates.id}")
+            .asUser(user)
+            .responseString()
+
+        val expected = draftWithFutureDates.copy(
+            decisionNumber = 1,
+            status = FeeDecisionStatus.SENT,
+            approvedBy = PersonData.JustId(testDecisionMaker_1.id),
+            decisionHandler = PersonData.JustId(testDecisionMaker_1.id),
+            documentKey = "feedecision_${draftWithFutureDates.id}_fi.pdf"
+        )
+        val actual = deserializeResult(result.get()).data
+        assertEqualEnough(expected.let(::toDetailed), actual)
+        assertEquals(actual.sentAt?.truncatedTo(ChronoUnit.DAYS), nowDate.atStartOfDay().toInstant(ZoneOffset.UTC))
```

Comment:
If you used mocked time in this test, you could just do a normal equality check. Like this:

```
val now = HelsinkiDateTime.of(...)
...
val (_, response, _) = http.post("/decisions/confirm")
    .withMockedTime(now)
...
assertEquals(now.toInstant(), actual.sentAt)
```